File: cypress/support/scripts/pyfaas4i/modeling/error/test-ts294.py
import pandas as pd
from pyfaas4i.faas import run_models

# ------ Define properties of the series -----------------
project_name = "ts294"

df_example1 = pd.read_excel("cypress/fixtures/Dataset/Trimestral/cagedServicos_RJ.xlsx")
    
# ------ Declare the date variable and its format --------

date_variable = 'data_tidy'
date_format = '%Y-%m-%d'

# ------ Dataframes must be passed in a dictionary
# ------ The key must be named after the variable of interest
data_list = {'saldo_cagedservico_RJ': df_example1}

# ------ Define FaaS parameters ---------------------------
model_spec = {
        'log': True,
        'seas.d': True,
        'n_steps': 4,
        'n_windows': 12,
        'n_best': 20,
        'accuracy_crit': 'MAPE',
        'info_crit': 'AIC',
        'exclusions': [['IDV_RJ_servicos_industriais']],
        'golden_variables': [],
        'fill_forecast': True,
        'cv_summary': 'mean',
        'selection_methods': {
            'lasso': True,
            'rf': True,
            'corr': True,
            'apply.collinear': ["corr","rf","lasso","no_reduction"]
        },
        'lags': {"IDV_RJ_servicos_industriais": [1,2,3], 
                 "IDV_RJ_outros_Servicos": [1,2,3],
                 "z_PIB_servicos_RJ": [1,2,3]},
        'allowdrift': True
}

run_models(data_list, date_variable, date_format, model_spec, project_name, version_check = False)

File: cypress/support/scripts/faas4i/modeling/error/test-TS294.R
library(faas4i)

dataset_1 <- readxl::read_excel("cypress/fixtures/Dataset/Mensal/dataset_1627.xlsx")
data_list_ex1 <-list(dataset_1)
names(data_list_ex1) <- c(names(dataset_1)[2])

# Also, specify the date variable and its format
date_variable <- "data_tidy"
date_format <- '%Y-%m-%d' # or'%m/%d/%Y'

# 3) Set Project Name
project_name <- "ts294"

model_spec_ex4 <- list(n_steps = 12,
                       n_windows = 2,
                       exclusions = list("fs_massa_real",
                                              "fs_rend_medio"),
                       selection_methods = list(apply.collinear = F))


# Send request
faas4i::run_models(data_list = data_list_ex1, date_variable = date_variable,
                   date_format = date_format, model_spec = model_spec_ex4,
                   project_name = project_name, version_check=F)